### ISSUE TITLE
[Bug] NPE/ClassCastException on select

### ISSUE THREAD

**Issue by matanshukry** [orientechnologies/orientdb#4966]
This exception seems to occur on conflicting fields. I do seems to recall a time when selecting multiple fields with the same name would result in an error, but it doesn't seems to be case.
This is tested against the develop branch (at the current time of writing this).
### Preparations

```
CREATE CLASS Animal EXTENDS V
CREATE PROPERTY Animal.id INTEGER
CREATE PROPERTY Animal.animalName STRING

CREATE CLASS Person EXTENDS V
CREATE PROPERTY Person.id INTEGER
CREATE PROPERTY Person.name STRING
CREATE PROPERTY Person.pet LINK Animal

CREATE VERTEX Animal SET id=0, name="Raxi"
CREATE VERTEX Person SET id=0, name="Joe", pet=(SELECT FROM Animal WHERE id=0)
```

Problematic query:

```
SELECT *,pet.name from Person
```

The problem seems to come a conflicting field named 'pet' and the type.
- include a field named 'pet' with type OIdentifiable (LINK to Animal)
  pet.name include a field named 'pet' (named automatically) with type String.

Changing the query by adding an alias, does not cause the issue (it is if you give the alias the same name, though).
## Client issues

After querying (using console), the client either:
1. Has no result, query is simply stuck
2. See the message Error: java.lang.NullPointerException , in which case a second query (doesn't matter what the query is, irrelevant to the first query) is stuck.
3. Gets a result without field, and like (2) above, second query is stuck regardless of the query.
## Server

The server log shows the following:

```
{db=test} Error on unmarshalling record #18:0 (java.lang.ClassCastException: java.lang.String cannot be cast to com.orientechnologies.orient.core.db.record.OIdentifiable)
java.lang.ClassCastException: java.lang.String cannot be cast to com.orientechnologies.orient.core.db.record.OIdentifiable
    at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.writeSingleValue(ORecordSerializerBinaryV0.java:568)
    at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.serialize(ORecordSerializerBinaryV0.java:257)
    at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinary.toStream(ORecordSerializerBinary.java:92)
    at com.orientechnologies.orient.core.record.impl.ODocument.toStream(ODocument.java:2313)
    at com.orientechnologies.orient.core.record.impl.ODocument.toStream(ODocument.java:660)
    at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.getRecordBytes(OBinaryNetworkProtocolAbstract.java:415)
    at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.writeRecord(OBinaryNetworkProtocolAbstract.java:430)
    at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.writeIdentifiable(OBinaryNetworkProtocolAbstract.java:135)
    at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.serializeValue(ONetworkProtocolBinary.java:1467)
    at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1415)
    at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:399)
    at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:217)
    at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

**Comment by maggiolo00** [comment]
hi @matanshukry  which version of OrientDB

**Comment by matanshukry** [comment]
As stated, I tested this against the latest version of develop branch.
I'm also seeing it on version 2.1-rc6

**Comment by tglman** [comment]
tested this, it's working correctly in 2.1.x but not on the development( 2.2.0-SNAPSHOT ), moving on the correct milestone.

**Comment by tglman** [comment]
Hi i tested it again in 2.2, it's still persist, but cannot be solved if not breaking the behaviour of the current sql result sets, as a workaround is sufficient to use an alias with a name that not match any fields of the record, ex : `SELECT *,pet.name  as petName from Person`

**Comment by tglman** [comment]
Hi,

This is fixed in 3.0 with a new result set API that provide that cannot be used to save directly the result document of a query.

Closing
